pkg/reconciler/taskrun/taskrun.go

```diff
@@ -813,6 +815,11 @@ func isTaskRunValidationFailed(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "TaskRun validation failed")
 }
 
+func isPodAdmissionFailed(err error) bool {
+	return err != nil && k8serrors.IsForbidden(err) && (strings.Contains(err.Error(), "violates PodSecurity") ||
+		strings.Contains(err.Error(), "security context constraint"))
```

Comment:
Using `strings.Contains` feels a little brittle since if the string content changes then it could break this logic. Is there a way to compare it better using `errors.Is`?

Reply:
Actually, inside `k8serrors.IsForbidden(err)`, `errors.As` has been used to convert err to [StatusError](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L35). 

In this case, StatusError.StatusReason is `Forbidden`, StatusError.Code is 403 and Status.Message shows the detail of error. The StatusError is from webhook as REST API server response which is the source of error. 

Using `strings.Contains` seems to be a common way to identify error. As you can see, other funcs in this file such as `isExceededResourceQuotaError`, `isTaskRunValidationFailed` are also using `strings.Contains`.

LBNL, we can't find another field from StatusError to identify such similar error cases.